doc/sphinx/language/core/basic.rst

```diff
@@ -527,20 +527,20 @@ The :cmd:`Set` and :cmd:`Unset` commands support the :attr:`local`,
 :attr:`global` and :attr:`export` locality attributes:
 
 * no attribute: the original setting is *not* restored at the end of
-  the current module or section.
+  the current section but it is restored at the end of the current module.
 * :attr:`local` (or alternatively, the ``Local`` prefix): the setting
   is applied within the current module or section.  The original value
   of the setting is restored at the end of the current module or
   section.
-* :attr:`export` (or alternatively, the ``Export`` prefix): similar to
-  :attr:`local`, the original value of the setting is restored at the
-  end of the current module or section.  In addition, if the value is
-  set in a module, then :cmd:`Import`\-ing the module sets the option
-  or flag.
+* :attr:`export` (or alternatively, the ``Export`` prefix): the
+  original setting is *not* restored at the end of the current section
+  but it is restored at the end of the current module. In addition,
+  the setting persists when :cmd:`Import`\ing the module in which it
+  is set.
 * :attr:`global` (or alternatively, the ``Global`` prefix): the
   original setting is *not* restored at the end of the current module
   or section.  In addition, if the value is set in a file, then
-  :cmd:`Require`\-ing the file sets the option.
+  :cmd:`Require`\-ing (even transitively) the file sets the option.
```

Comment:
```suggestion
  :cmd:`Require`\-ing the file (even transitively) sets the option.
```

The language in this section is confusing:
- we have "settings", which are flags and options, both of which can be "set" and "unset".
- "sets the option" doesn't seem to include flags
- if you unset a flag, can the "false" be exported?  If so, "if the value is set in a file" is misleading

I think we need a word that means "option or flag" other than "setting".  Or just use that phrase
repeatedly for clarity.  Maybe use "value" in place of some uses of "setting".

Reply:
>if you unset a flag, can the "false" be exported?

Yes.

Note that "setting" also includes table. Otherwise, option would be a generic term (flags are boolean options).

I will try to improve the wording.

Reply:
Yes, "settings" includes tables, but the `Set` and `Unset` commands don't apply to tables, which use `Add` and `Remove` instead.  That's a little confusing.

Reply:
Indeed. And you just made me dig what was the behavior for `Add` and `Remove` and I was surprised to discover it is actually different (the default is `local` and `export` is not supported):

```coq
Section section.

  Add Search Blacklist "foo".

  Test Search Blacklist for "foo".
  (* Search blacklist does include foo *)

End section.

Test Search Blacklist for "foo".
(* Search blacklist does not include foo *)
```

Reply:
I've changed the text to use the wording "flag or option" instead of "setting".